wgpu-hal/src/metal/mod.rs

```diff
@@ -124,30 +139,47 @@ impl crate::Instance for Instance {
         _display_handle: raw_window_handle::RawDisplayHandle,
         window_handle: raw_window_handle::RawWindowHandle,
     ) -> Result<Surface, crate::InstanceError> {
-        match window_handle {
-            #[cfg(any(target_os = "ios", target_os = "visionos"))]
-            raw_window_handle::RawWindowHandle::UiKit(handle) => {
-                Ok(unsafe { Surface::from_view(handle.ui_view.cast()) })
+        let layer = match window_handle {
+            raw_window_handle::RawWindowHandle::AppKit(handle) => unsafe {
+                raw_window_metal::Layer::from_ns_view(handle.ns_view)
+            },
+            raw_window_handle::RawWindowHandle::UiKit(handle) => unsafe {
+                raw_window_metal::Layer::from_ui_view(handle.ui_view)
+            },
+            _ => {
+                return Err(crate::InstanceError::new(format!(
+                    "window handle {window_handle:?} is not a Metal-compatible handle"
+                )))
             }
-            #[cfg(target_os = "macos")]
-            raw_window_handle::RawWindowHandle::AppKit(handle) => {
-                Ok(unsafe { Surface::from_view(handle.ns_view.cast()) })
-            }
-            _ => Err(crate::InstanceError::new(format!(
-                "window handle {window_handle:?} is not a Metal-compatible handle"
-            ))),
-        }
+        };
+
+        // SAFETY: The layer is an initialized instance of `CAMetalLayer`, and
+        // we transfer the retain count to `Retained` using `into_raw`.
+        let layer = unsafe {
+            Retained::from_raw(layer.into_raw().cast::<CAMetalLayer>().as_ptr()).unwrap()
+        };
+
+        Ok(Surface::new(layer))
     }
 
     unsafe fn enumerate_adapters(
         &self,
         _surface_hint: Option<&Surface>,
     ) -> Vec<crate::ExposedAdapter<Api>> {
-        let devices = metal::Device::all();
+        // MTLCopyAllDevices is always available on macOS and Mac Catalyst,
+        // but only available recently on iOS 18.0 / tvOS 18.0 / visionOS 2.0.
+        // Instead, we use MTLCreateSystemDefaultDevice on those platforms.
+        //
+        // TODO: Call MTLCopyAllDevices once objc2-metal does this internally:
+        // https://github.com/madsmtm/objc2/commit/35439408e684adddc54d7ce954ae2661716d9ef8
+        #[cfg(any(target_os = "macos", target_abi = "macabi"))]
+        let devices = objc2_metal::MTLCopyAllDevices(); // NSArray -> into_iter
+        #[cfg(not(any(target_os = "macos", target_abi = "macabi")))]
+        let devices = objc2_metal::MTLCreateSystemDefaultDevice(); // Option -> into_iter
```

Comment:
question: Is there a way to track this via an issue?

Reply:
It was fixed in v0.3.1, just forgot to update it here.